code/datums/actions/xeno_action.dm

```diff
@@ -1,86 +1,32 @@
 
-/datum/action/xeno_action
+/datum/action/ability/xeno_action
 	///If you are going to add an explanation for an ability. don't use stats, give a very brief explanation of how to use it.
 	desc = "This ability can not be found in codex."
 	action_icon = 'icons/Xeno/actions.dmi'
-	var/plasma_cost = 0
-	///bypass use limitations checked by can_use_action()
-	var/use_state_flags = NONE
-	var/last_use
-	var/cooldown_timer
-	var/ability_name
-	var/keybind_flags
-	var/cooldown_id
-	var/target_flags = NONE
-	/// flags to restrict a xeno ability to certain gamemode
-	var/gamemode_flags = ABILITY_ALL_GAMEMODE
 
-/datum/action/xeno_action/New(Target)
+/datum/action/ability/xeno_action/New(Target)
```

Comment:
Renaming everythong to xeno_ability ? Or just ability/xeno

Reply:
just made it ability/xeno